pymc/backends/ndarray.py

```diff
@@ -108,6 +108,8 @@ def record(self, point, sampler_stats=None) -> None:
         samples = self.samples
         draw_idx = self.draw_idx
         for varname, value in zip(self.varnames, self.fn(*point.values())):
+            print(f"DEBUG: draw_idx={draw_idx}, max_index={samples[varname].shape[0]}")
+            print(f"DEBUG: samples shape = {samples[varname].shape}")
```

Comment:
We don't want print statements in the code.

If anything, `logging.getLogger().debug()` should be used, but also not in this method which gets called thousands of times per second.